guides/plugins/plugins/checkout/payment/add-payment-plugin.md

```diff
@@ -27,13 +27,16 @@ You can create your own payment handler by implementing one of the following int
 * `finalize`: The `finalize` method is only required if you implemented the `AsynchronousPaymentHandlerInterface`, returned a `RedirectResponse` in your `pay` method and the customer has been redirected from the payment provider back to Shopware 6. You must check here if the payment was successful or not and update the order transaction state accordingly. Similar to the pay action you are able to throw exceptions if some error cases occur. Throw the `CustomerCanceledAsyncPaymentException` if the customer canceled the payment process on the payment provider site. If another general error occurs throw the `AsyncPaymentFinalizeException` e.g. if your call to the payment provider API fails. Shopware 6 will handle these exceptions and will set the transaction to the `cancelled` state.
+* `validate`: This method will be called before an order is placed and should check, if a given prepared payment is valid. The payment handler has to verify the given payload with the payment service, because Shopware cannot ensure that the transaction created by the frontend is valid for the current cart. Throw an `ValidatePreparedPaymentException` to fail the validation in your implementation.
+* `capture`: Will be called after successfully validating a prepared payment in the `validate` method. After successful verification the order will be created and the payment handler will be called again to charge the payment. When the charge was successful the payment will be set to paid and the user will be forwarded to the finish page. Throw an `CapturePreparedPaymentException` on any errors to fail the capture process and the after order process will be active and the customer can complete the payment again.
```

Comment:
I'm not happy with this description. The status will not be set to paid, unless you do it yourself in the prepared payment handler.

Also I would make it more clear in the first sentence, that this not only is executed after validation but also after order creation.